src/prog_models/models/uav_model/uav_model.py

```diff
@@ -0,0 +1,556 @@
+
+    :term:`States<state>`: (13)
+        | x: cartesian position
+        | y: cartesian position
+        | z: cartesian position
+        | phi: pitch
+        | theta: roll
+        | psi: yaw
+        | vx: velocity 
+        | vy: velocity 
+        | vz: velocity 
+        | p: angular velocity 
+        | q: angular velocity 
+        | r: angular velocity 
+        | t: time 
+
+    :term:`Outputs<output>`: (12)
+        | x: cartesian position
+        | y: cartesian position
+        | z: cartesian position
+        | phi: pitch
+        | theta: roll
+        | psi: yaw
+        | vx: velocity 
+        | vy: velocity 
+        | vz: velocity 
+        | p: angular velocity 
+        | q: angular velocity 
+        | r: angular velocity 
+        | t: time 
+
+    Keyword Args
+    ------------
+        process_noise : Optional, float or dict[str, float]
+          :term:`Process noise<process noise>` (applied at dx/next_state). 
+          Can be number (e.g., .2) applied to every state, a dictionary of values for each 
+          state (e.g., {'x1': 0.2, 'x2': 0.3}), or a function (x) -> x
+        process_noise_dist : Optional, str
+          distribution for :term:`process noise` (e.g., normal, uniform, triangular)
+        measurement_noise : Optional, float or dict[str, float]
+          :term:`Measurement noise<measurement noise>` (applied in output eqn).
+          Can be number (e.g., .2) applied to every output, a dictionary of values for each
+          output (e.g., {'z1': 0.2, 'z2': 0.3}), or a function (z) -> z
+        measurement_noise_dist : Optional, str
+          distribution for :term:`measurement noise` (e.g., normal, uniform, triangular)
+        flight_file : Optional, str
+          Text file to specify waypoint information. Necessary columns must be in the following
+          order with units specified: latitude (labeled 'lat_deg' or 'lat_rad'), longitude 
+          (labeled 'lon_deg' or 'lon_rad'), and altitude (labeled 'alt_ft' or 'alt_m'). An 
+          additional column for ETAs may be included (labeled 'time_unix). Note that while
+          'flight_file' is optional, either 'flight_file' or 'flight_plan' must be specified.
+        flight_plan : Optional, dict[str, numpy array]
+          Dictionary to specify waypoint information. Necessary keys must include the following
+          with units specified: latitude ('lat_deg' or 'lat_rad'), longitude 
+          (labeled 'lon_deg' or 'lon_rad'), and altitude (labeled 'alt_ft' or 'alt_m'). An 
+          additional key for ETAs may be included (labeled 'time_unix). Each key must correspond
+          to a numpy array of values. Note that while 'flight_plan' is optional, either 
+          'flight_file' or 'flight_plan' must be specified.
+        flight_name : Optional, str
+          Optional string to identify flight plan.
+        aircraft_name : Optional, str
+          Optional string to identify aircraft. 
+        dt : float
```

Comment:
These and all the params below are optional as well, right?

Reply:
Most are, good catch. The speeds are funny because they are either *not* an option if ETAs are defined (any input values are ignored and warning is printed) or *required* if no ETAs are defined. Do you think these should be listed as Optional?

Reply:
Hmmm... good question. I would say, lets leave the optional flag off of those